spec/facility_spec.rb

```diff
@@ -1,26 +1,181 @@
+      registrant_1 = Registrant.new('Bruce', 18, true)
+      registrant_2 = Registrant.new('Penny', 16)
+      registrant_3 = Registrant.new('Tucker', 15)
+
+      facility_1.administer_written_test(registrant_1)
+      facility_1.administer_road_test(registrant_1)
+
+      expect(facility_1.administer_road_test(registrant_1)).to be true
+    end
+
+    it "can renew one's license" do
+      facility_1 = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      facility_1.add_service('Written Test')
+      facility_1.add_service('Road Test')
+      facility_1.add_service('Renew License')
+      registrant_1 = Registrant.new('Bruce', 18, true)
+      registrant_2 = Registrant.new('Penny', 16)
+      registrant_2.earn_permit
+      registrant_3 = Registrant.new('Tucker', 15)
+
+      facility_1.administer_written_test(registrant_1)
+      facility_1.administer_road_test(registrant_1)
+      facility_1.renew_drivers_license(registrant_1)
+      facility_1.administer_written_test(registrant_2)
+      facility_1.administer_road_test(registrant_2)
+      facility_1.administer_written_test(registrant_3)
+      facility_1.administer_road_test(registrant_3)
+
+      expect(registrant_1.license_data).to eq({:written=>true, :license=>true, :renewed=>true})
+    end
+
+    it "has access to DMV Colorado locations" do
+      colorado_facilities = DmvDataService.new.co_dmv_office_locations
+      new_york_facilities = DmvDataService.new.ny_dmv_office_locations
+      missouri_facilities = DmvDataService.new.mo_dmv_office_locations
+
+      expect(colorado_facilities).to be_a(Array)
+    end
+
+    it "has access to DMV New York locations" do
+      colorado_facilities = DmvDataService.new.co_dmv_office_locations
+      new_york_facilities = DmvDataService.new.ny_dmv_office_locations
+      missouri_facilities = DmvDataService.new.mo_dmv_office_locations
+
+      expect(new_york_facilities).to be_a(Array)
+    end
+
+    it "has access to DMV Missouri locations" do
+      colorado_facilities = DmvDataService.new.co_dmv_office_locations
+      new_york_facilities = DmvDataService.new.ny_dmv_office_locations
+      missouri_facilities = DmvDataService.new.mo_dmv_office_locations
+
+      expect(missouri_facilities).to be_a(Array)
+    end
+
+    it "can create facility instances from DMV data" do
+      facility_list = Facility.new({})
+      colorado_facilities = DmvDataService.new.co_dmv_office_locations
+      new_york_facilities = DmvDataService.new.ny_dmv_office_locations
+      missouri_facilities = DmvDataService.new.mo_dmv_office_locations
+
+      facility_list.create_facility(colorado_facilities)
+      facility_list.create_facility(new_york_facilities)
+      facility_list.create_facility(missouri_facilities)
+
+      expect(facility_list.create_facility(colorado_facilities)).to be_a(Array)
+    end
```

Comment:
For tests like this, I'd also suggest going into that array a bit and testing that the elements are in fact fAcility objects and that their attributes are not nil.

For starters, you could just pull the first element out of the array and test that it's the right kind of object, and that when you call to get the attributes, you aren't getting returned `nil`.